utils/s3/client.cc

```diff
@@ -766,7 +766,10 @@ future<> client::multipart_upload::abort_upload() {
     s3l.trace("DELETE upload {}", _upload_id);
     auto req = http::request::make("DELETE", _client->_host, _object_name);
     req.query_parameters["uploadId"] = std::exchange(_upload_id, ""); // now upload_started() returns false
-    co_await _client->make_request(std::move(req), ignore_reply, http::reply::status_type::no_content);
+    co_await _client->make_request(std::move(req), ignore_reply, http::reply::status_type::no_content).handle_exception([](const auto&) -> future<> {
+        // After all retries failed return success, dangling parts will be removed by other means - `rclone cleanup` or S3 bucket's Lifecycle Management Policy
```

Comment:
i *guess* i understand what you mean here:

when an failed upload, we abort the failing upload by DELETE'ing the partially uploaded object. but if the DELETE operation fails, instead of populating the failure to the caller, we swallow it, and expect either `rclone cleanup` or S3 bucket's Lifecycle Management Policy to take care of this.

anyway, could you please expand the commit message, and the comment a little bit, so they are more readable. and, might want to log the upload id and the object's key before returning.